tools/checkpoint_conversion/convert_embedding_gemma_checkpoints.py

```diff
@@ -0,0 +1,131 @@
+"""
+Convert a pre-trained causal Gemma3 Keras model to an Embedding Gemma model.
+
+This script takes a standard Gemma3 model designed for causal language modeling
+and adapts it for sentence embedding tasks. It modifies the model architecture
+for bi-directional attention, adds a new pooling head for generating fixed-size
+embeddings, and transfers the weights from the original model.
+
+Setup:
+```shell
+# Make sure to install the necessary libraries, including the specific
+# keras_hub package containing the Gemma3 models.
+pip install keras-hub
+pip install keras
+
+Usage:
+```shell
+cd tools/checkpoint_conversion
+python3 convert_embedding_gemma_checkpoints.py \
+    --source_preset gemma3_instruct_4b_text \
+    --output_preset embedding_gemma3_4b_en \
+    --pooling_intermediate_dim 4096 \
+    --embedding_dim 768
+"""
+
+import argparse
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
+os.environ["CUDA_VISIBLE_DEVICES"] = "-1"
+
+import keras
+
+from keras_hub.src.models.gemma3.gemma3_backbone import Gemma3Backbone
+from keras_hub.src.models.gemma3.gemma3_tokenizer import Gemma3Tokenizer
+
+
```

Comment:
Can we have `PRESET_MAP` like we have for other converters for all the models we want to convert it to embedding model, it will be easier to map and track for any future version uploading to Kaggle.

Something like

```
PRESET_MAP = {
    "embedding_gemma3_4b_en": "gemma3_instruct_4b_text",
    ...
}
```